### ISSUE TITLE
`doc-paragraphs-missing-punctuation` destroys function name

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#16169]
Using the following flags

`--force-warn clippy::doc-paragraphs-missing-punctuation`

this code:

```rust
#![allow(unused)]
#![warn(clippy::empty_docs)]
mod outer {
    //!


    ///
    mod inner {
        ///
        fn dont_warn_inner_outer() {
            //!w
        }

    }


}

```

caused the following diagnostics:

```
    Checking _a v0.1.0 (/tmp/icemaker_global_tempdir.whL0Hzp6OLAl/icemaker_clippyfix_tempdir.zp3qQo2WqkRz/_a)
warning: doc paragraphs should end with a terminal punctuation mark
  --> src/lib.rs:10:18
   |
10 |         fn dont_warn_inner_outer() {
   |                  ^ help: end the paragraph with some punctuation: `.`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_paragraphs_missing_punctuation
   = note: requested on the command line with `--force-warn clippy::doc-paragraphs-missing-punctuation`

warning: `_a` (lib) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.16s

```

However after applying these diagnostics, the resulting code:

```rust
#![allow(unused)]
#![warn(clippy::empty_docs)]
mod outer {
    //!


    ///
    mod inner {
        ///
        fn dont_w().arn_inner_outer() {
            //!w
        }

    }


}

```
no longer compiled:

```
    Checking _a v0.1.0 (/tmp/icemaker_global_tempdir.whL0Hzp6OLAl/icemaker_clippyfix_tempdir.zp3qQo2WqkRz/_a)
error: expected one of `->`, `where`, or `{`, found `.`
  --> src/lib.rs:10:20
   |
10 |         fn dont_w().arn_inner_outer() {
   |                    ^ expected one of `->`, `where`, or `{`

error: could not compile `_a` (lib test) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `_a` (lib) due to 1 previous error

```

Version:
```
rustc 1.93.0-nightly (9b82a4fff 2025-12-01)
binary: rustc
commit-hash: 9b82a4fffe2b215f488a6dfdc0b508235f37c85c
commit-date: 2025-12-01
host: x86_64-unknown-linux-gnu
release: 1.93.0-nightly
LLVM version: 21.1.5

```
